test/Microsoft.AspNet.Authentication.Test/Facebook/FacebookMiddlewareTests.cs

```diff
@@ -62,6 +64,37 @@ public class FacebookMiddlewareTests
         }
 
         [Fact]
+        public async Task MapWillNotAffectRedirect()
```

Comment:
Is it worth adding another test where `app.UseFacebookAuthentication` and `app.Map` are both in a global `app.Map`?.

```csharp
var server = CreateServer(app => {
    app.Map("/base", map => {
        map.UseFacebookAuthentication();
        map.Map("/login", signoutApp => signoutApp.Run(context => context.Authentication.ChallengeAsync("Facebook", new AuthenticationProperties() { RedirectUri = "/" })));
     });
});
```

In this case, the `redirect_uri` should be `http://example.com/base/signin-facebook`.